cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Gov/NoConfidence.hs

```diff
@@ -0,0 +1,263 @@
+      ]
+
+  committeeVkey1Fp <- H.noteShow $ work </> defaultCommitteeVkeyFp 1
+
+  -- Read committee cold keys from disk to put into conway genesis
+
+  comKeyHash1Str <- filter (/= '\n') <$> H.execCli' execConfigOffline
+      [ anyEraToString cEra, "governance", "committee"
+      , "key-hash"
+      , "--verification-key-file", committeeVkey1Fp
+      ]
+
+  CommitteeColdKeyHash comKeyHash1 <-
+    H.evalEither
+      $ deserialiseFromRawBytesHex (AsHash AsCommitteeColdKey)
+      $ BSC.pack comKeyHash1Str
+
+  let comKeyCred1 = L.KeyHashObj comKeyHash1
+      committeeThreshold = unsafeBoundedRational 0.5
+      committee = L.Committee (Map.fromList [(comKeyCred1, EpochNo 100)]) committeeThreshold
+
+  alonzoGenesis <- evalEither $ first prettyError defaultAlonzoGenesis
+  (startTime, shelleyGenesis') <- getDefaultShelleyGenesis fastTestnetOptions
+  let conwayGenesisWithCommittee =
+        defaultConwayGenesis { L.cgCommittee = committee }
+
+  TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    , wallets=wallet0:_wallet1:_
+    , configurationFile
+    } <- cardanoTestnet
+           fastTestnetOptions
+           conf startTime shelleyGenesis'
+           alonzoGenesis conwayGenesisWithCommittee
+
+  poolNode1 <- H.headM poolNodes
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1 testnetMagic
+
+  let socketName' = IO.sprocketName poolSprocket1
+      socketBase = IO.sprocketBase poolSprocket1 -- /tmp
+      socketPath = socketBase </> socketName'
+
+  _epochStateView <- getEpochStateView (File configurationFile) (File socketPath)
+
+  H.note_ $ "Sprocket: " <> show poolSprocket1
+  H.note_ $ "Abs path: " <> tempAbsBasePath'
+  H.note_ $ "Socketpath: " <> socketPath
+
+  res <- findCondition (committeeIsPresent False) configurationFile socketPath (EpochNo 10)
+  mRes <- evalEither res
+  H.assert $ isJust mRes
+
+  -- Step 2. Propose motion of no confidence. DRep and SPO voting thresholds must be met.
+
+  -- Create proposal to add a new member to the committee
+
+  proposalAnchorFile <- H.note $ work </> "sample-proposal-anchor"
+  H.writeFile proposalAnchorFile "dummy anchor data"
+
+  proposalAnchorDataHash <- H.execCli' execConfig
+    [ eraToString era, "governance"
+    , "hash", "anchor-data", "--file-text", proposalAnchorFile
+    ]
+
+
+  proposalFile <- H.note $ work </> "sample-proposal-anchor"
+  stakeVkeyFp <- H.note $ work </> "stake.vkey"
+  stakeSKeyFp <- H.note $ work </> "stake.skey"
+
+  _ <- P.cliStakeAddressKeyGen tempAbsPath'
+         $ P.KeyNames { P.verificationKeyFile = stakeVkeyFp
+                      , P.signingKeyFile = stakeSKeyFp
+                      }
+  void $ H.execCli' execConfig $
+    [ eraToString era, "governance", "action", "create-no-confidence"
+    , "--testnet"
+    , "--governance-action-deposit", show @Integer 1_000_000 --- TODO: Get from protocol parameters
```

Comment:
You can get the value from epoch state view: https://github.com/IntersectMBO/cardano-node/blob/6c36c4d6e9923d2b8fb7ad36364c734392bcf7b9/cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Gov/DRepActivity.hs#L274

Reply:
Thats retrieves _drep_ deposit not a governance action deposit.

Reply:
Ah right. So `ppGovActionDepositL` should be used instead.

Reply:
I've wrote this in `Testnet.Components.Query` for my other PR I'm working on. Feel free to use it if you prefer.
```haskell
-- | Obtain minimum deposit amount for governance action from node
getMinGovActionDeposit
  :: HasCallStack
  => MonadAssertion m
  => MonadIO m
  => MonadTest m
  => EpochStateView
  -> ConwayEraOnwards era
  -> m Integer -- ^ The governance state
getMinGovActionDeposit epochStateView ceo = withFrozenCallStack $ do
  govState <- getGovState epochStateView ceo
  pure $ conwayEraOnwardsConstraints ceo $ govState ^. L.cgsCurPParamsL . L.ppGovActionDepositL  . to L.unCoin
```